modules/packages/Python.chpl

```diff
@@ -1151,7 +1155,21 @@ module Python {
       return pyDict;
     }
 
-    // TODO: convert chapel map to python dict
+    /*
+      Convert a chapel map to a python dict.  Returns a new reference
+    */
+    @chpldoc.nodoc
+    proc toDict(m: Map.map(?)): PyObjectPtr throws {
+      var pyDict = PyDict_New();
+      this.checkException();
+      for key in m.keys() {
+        var pyKey = toPython(key);
+        var pyValue = toPython(m[key]);
+        PyDict_SetItem(pyDict, pyKey, pyValue);
```

Comment:
I think you need a decref for pyValue, see https://docs.python.org/3/c-api/dict.html#c.PyDict_SetItem